### ISSUE TITLE
volume create brick directory validation logic is incorrect

### ISSUE THREAD

**Issue by atinmu** [gluster/glusterd2#1384]
### Observed behavior

root@b9bad6008a5e:/home/go/src/github.com/gluster/glusterd2/build# ./glustercli volume create test-vol21 replica 3 172.17.0.2:/tmp/b21 172.17.0.3:/tmp/21 172.17.0.4:/tmp/b21 --force; ./glustercli volume start test-vol$i;
Volume creation failed

Response headers:
X-Gluster-Cluster-Id: 8d9dae2d-b422-476a-aa5d-5f7e6cb5eb1d
X-Gluster-Peer-Id: 75a41e07-8722-427c-bd8c-89e5cb694447
X-Request-Id: feb7d1e8-dfee-445c-b7b1-114f01dbcd78

Response body:
Transaction step vol-create.ValidateBricks failed on peer 75a41e07-8722-427c-bd8c-89e5cb694447 with error: path /tmp/b21 is a subdirectory of another existing brick with path /tmp/b2 belonging to volume (name=test-vol2;id=d272db49-e889-41de-a147-8ad5ef939360)
Transaction step vol-create.ValidateBricks failed on peer daefdcb1-2fc5-4e06-a4e8-251c8d29e6d3 with error: path /tmp/b21 is a subdirectory of another existing brick with path /tmp/b2 belonging to volume (name=test-vol2;id=d272db49-e889-41de-a147-8ad5ef939360)

volume start failed

Response headers:
X-Gluster-Cluster-Id: 8d9dae2d-b422-476a-aa5d-5f7e6cb5eb1d
X-Gluster-Peer-Id: 75a41e07-8722-427c-bd8c-89e5cb694447
X-Request-Id: 13dd413a-4878-4969-b00a-48bf821cb7d1

Response body:
volume already started

Two problems I see here:

1. Why did response body reported that volume already started? 
2. /tmp/b20 isn't inside /tmp/b2 but validation logic thinks otherwise.

### Expected/desired behavior

volume should be created and started successfully.

### Information about the environment:

- Glusterd2 version used ( master): 
- Operating system used: 
- Glusterd2 compiled from sources, as a package (rpm/deb), or container: 
- Using External ETCD: (yes/no, if yes ETCD version): yes , 3.2.2
- If container, which container image: 
standalone

### Other useful information

- glusterd2 config files from all nodes (default /etc/glusterd2/glusterd2.toml)
- glusterd2 log files from all nodes (default /var/log/glusterd2/glusterd2.log)
- ETCD configuration
- Contents of `uuid.toml` from all nodes (default /var/lib/glusterd2/uuid.toml)
- Output of `statedump` from any one of the node

### Useful commands

- To get glusterd2 version
    ```
    glusterd2 --version
    ```
- To get ETCD version
    ```
    etcd --version
    ```
- To get output of statedump
    ```
    curl http://glusterd2-IP:glusterd2-Port/statedump
    ```

**Comment by Madhu-1** [comment]
>root@b9bad6008a5e:/home/go/src/github.com/gluster/glusterd2/build# ./glustercli volume create test-vol21 replica 3 172.17.0.2:/tmp/b21 172.17.0.3:/tmp/21 172.17.0.4:/tmp/b21 --force; ./glustercli volume start test-vol$i;

@atinmu is the above script is complete? can you provide the script used for your testing

**Comment by atinmu** [comment]
@Madhu-1 

for i in {1..20}; do ./glustercli volume create test-vol$i replica 3 172.17.0.2:/tmp/b-$i 172.17.0.3:/tmp/b-$i 172.17.0.4:/tmp/b-$i --force; ./glustercli volume start test-vol$i; done

Run the above and you should be able to see for test-vol10 onwards the volume create requests would fail.

**Comment by Madhu-1** [comment]
@atinmu I was able to create and start volume from the above script with latest master code
```
[root@dhcp43-209 build]# for i in {1..20}; do ./glustercli volume create test-vol$i replica 3 10.70.42.71:/tmp/b-$i 10.70.43.106:/tmp/b-$i 10.70.43.209:/tmp/b-$i --force; ./glustercli volume start test-vol$i; done
test-vol1 Volume created successfully
Volume ID:  81dfc9ed-b6c4-4b6f-a339-dc6f1e9fc750
Volume test-vol1 started successfully
test-vol2 Volume created successfully
Volume ID:  90908431-b71c-4cb8-9ad4-d02641df15bb
Volume test-vol2 started successfully
test-vol3 Volume created successfully
Volume ID:  e7b87e8f-65d1-41e5-8da1-357f1fccf48a
Volume test-vol3 started successfully
test-vol4 Volume created successfully
Volume ID:  4a3b79b5-e587-4e00-ba53-3e4412b7f2a2
Volume test-vol4 started successfully
test-vol5 Volume created successfully
Volume ID:  15b62e62-65da-4ce5-b986-31d44a27bdd1
Volume test-vol5 started successfully
test-vol6 Volume created successfully
Volume ID:  830231e8-6e00-479e-a3e9-32cc00101f27
Volume test-vol6 started successfully
test-vol7 Volume created successfully
Volume ID:  fb865d2f-5cbb-4c7d-b8b5-6fb8da43adcd
Volume test-vol7 started successfully
test-vol8 Volume created successfully
Volume ID:  099a0a99-0f9c-4cb3-94eb-f393cb4359b1
Volume test-vol8 started successfully
test-vol9 Volume created successfully
Volume ID:  ad068872-7e9e-4416-b927-f88f38d2348f
Volume test-vol9 started successfully
test-vol10 Volume created successfully
Volume ID:  65d6de5d-821b-432d-9c9e-67d5174862fb
Volume test-vol10 started successfully
test-vol11 Volume created successfully
Volume ID:  65a11251-11c4-47f7-80e1-c42df735a551
Volume test-vol11 started successfully
test-vol12 Volume created successfully
Volume ID:  cce0b336-8e15-4d45-a404-6c953c61c25f
Volume test-vol12 started successfully
test-vol13 Volume created successfully
Volume ID:  b3abd28d-eb2d-4a54-a03f-83e14260a93b
Volume test-vol13 started successfully
test-vol14 Volume created successfully
Volume ID:  a4891a94-a20a-4468-bd09-987ed4047d28
Volume test-vol14 started successfully
test-vol15 Volume created successfully
Volume ID:  e09f632b-4d2f-4f9c-a4d5-8c970ec16dfe
Volume test-vol15 started successfully
test-vol16 Volume created successfully
Volume ID:  bcdc1807-e392-4fdf-bc21-0845607108fe
Volume test-vol16 started successfully
test-vol17 Volume created successfully
Volume ID:  9eedf5c0-a1d4-447e-9ac6-f503d57864c1
Volume test-vol17 started successfully
test-vol18 Volume created successfully
Volume ID:  2937f0ab-4200-4fe2-ba1d-09371aff9a65
Volume test-vol18 started successfully
test-vol19 Volume created successfully
Volume ID:  c697bb21-eaf5-4b03-ad7b-2cfe6635dbb6
Volume test-vol19 started successfully
test-vol20 Volume created successfully
Volume ID:  490575c1-0ea8-4625-9f5e-0ffe3c415048
Volume test-vol20 started successfully
[root@dhcp43-209 build]# 
```

**Comment by rishubhjain** [comment]
@atinmu I think that is because you are using shared space, I tried using a shared space and was able to reproduce the issue. 
Try:
`for i in {1..20}; do ./glustercli volume create test-vol$i replica 3 172.17.0.2:/tmp/b-$i 172.17.0.3:/tmp2/b-$i 172.17.0.4:/tmp1/b-$i  --force --create-brick-dir; ./glustercli volume start test-vol$i; done`

When I provided different path the issue was resolved. Do let me know if you are still facing the same issue.
